Automattic/newspack-plugin
# Linting: use lint-staged for JS pre-commit linting

### All Submissions:






### Changes proposed in this Pull Request:



`xwd-dev-lib` lints files in a tmp directory - with changed files only - which makes [`import/no-unresolved`](https://github.com/benmosher/eslint-plugin-import/blob/master/docs/rules/no-unresolved.md) rule fail at times, as not all files are present. 
This PR relieves `xwp-dev-lib` of linting JS files and makes it so that JS is linted with [`lint-staged`](https://github.com/okonet/lint-staged) tool.

## Important

Git hooks need to be updated after these changes: `./vendor/bin/cghooks update`

### How to test the changes in this Pull Request:

1. Update git hooks - `./vendor/bin/cghooks update`
2. Change a JS file so that it's broken (would not pass linting)
3. Try to commit the change. See that it fails and error message from Eslint is output. 

### Other information:
